pkg/blockstorage/zone/zone.go

```diff
@@ -83,20 +86,45 @@ func getAvailableZones(ctx context.Context, kubeCli kubernetes.Interface, validZ
 	sanitizedAvailableZones := SanitizeAvailableZones(availableZones, validZoneNames)
 	// Add all available valid source zones
 	for _, zone := range sourceZones {
-		z := getZoneFromKnownNodeZones(zone, sanitizedAvailableZones)
-		if z != "" {
+		if z := getZoneFromAvailableZones(zone, sanitizedAvailableZones); z != "" {
 			newZones[z] = struct{}{}
 		}
 	}
-	// If source zones aren't available and valid add all valid available zones
+	// If source zones aren't available get consistent zone from available zones
 	if len(newZones) == 0 {
-		for zone := range sanitizedAvailableZones {
-			newZones[zone] = struct{}{}
+		for _, zone := range sourceZones {
+			if z := consistentZone(zone, sanitizedAvailableZones); z != "" {
+				newZones[z] = struct{}{}
+			}
 		}
 	}
 	return newZones
 }
 
+// consistentZone will return the same zone given a source and a list of zones.
```

Comment:
Please also add some comments on why we need to do this, otherwise the next change to this implementation may also    revert this functionality.